### PR TITLE
Update best_practices.py

### PR DESCRIPTION
**Pull request by swapniljha001**
# I have made things!
* Contributes to issue #1525 
* Proofread and made changes from line 1730 to 2102.

## Checklist

- [x ] I have double-checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [x ] I have updated the documentation for the changes I have made

## Issues mentioned 
- Refs #1525

## Review needed
Some feedback on the proposed changes will help me in making more changes.

### CODE REVIEW

**Review by sobolevn — COMMENTED** [review]
Thanks a lot! 

CC @Dreamsorcerer

**Review by Dreamsorcerer — COMMENTED** [review]
FYI, there's no actual grammar fixes in here.
The changes mostly feel like personal preference.

To give an idea of the fixes we're really looking for, after quickly flicking through that page, here are a couple of examples of actual mistakes that should be fixed:
"Refactor your code to much [match] the style."
"Forbids to use anything rather [other] than..."
"These two violations look so similar." -> "These two exceptions look similar."

I don't think there's that many mistakes on this page though, probably more on the other pages.

File: wemake_python_styleguide/violations/best_practices.py [hunk 1]
```diff
@@ -1807,7 +1807,7 @@ class ApproximateConstantViolation(ASTViolation):
 
     Reasoning:
         Some constants are already defined.
-        No need to write to them again, use existing values.
+        No need to write them again, use existing values.
```
Comment (Dreamsorcerer) [inline wemake_python_styleguide/violations/best_practices.py:1810]: I misread this one first time. This is a legitimate grammar fix.

File: wemake_python_styleguide/violations/best_practices.py [hunk 2]
```diff
@@ -2075,15 +2075,15 @@ class ShebangViolation(SimpleViolation):
...
+        - Presence of whitespace before the shebang.
+        - Presence of blank lines or commented lines before the shebang.
 
     Reasoning:
         Setting the shebang incorrectly causes an executable mismatch.
 
     Solution:
-        Ensure the shebang is present on the first line,
-        contains "python", and there is no whitespace before.
+        Ensure that the shebang present on the first line,
```
[the remark this answers is not in the document]
Comment (swapniljha001) [reply wemake_python_styleguide/violations/best_practices.py:2085]: Ensure the shebang on the first line contains "Python"
Ensure the shebang is on the first line contains "Python".

I believe is is incorrect.

**Comment by swapniljha001** [comment]
> FYI, there's no actual grammar fixes in here. (Edit: there is 1)
> The changes mostly feel like personal preference.
> 
> To give an idea of the fixes we're really looking for, after quickly flicking through that page, here are a couple of examples of actual mistakes that should be fixed:
> "Refactor your code to much [match] the style."
> "Forbids to use anything rather [other] than..."
> "These two violations look so similar." -> "These two exceptions look similar."
> 
> I don't think there are that many mistakes on this page though, probably more on the other pages.

Can you tell me the number of lines, where these erroneous codes are?
I only went through the last couple hundred lines of code. I might have missed these.

**Comment by Dreamsorcerer** [comment]
I was just scanning through on the website: https://wemake-python-stylegui.de/en/latest/pages/usage/violations/best_practices.html#best-practices
Ctrl+F for a couple of words should find them.

File: wemake_python_styleguide/violations/best_practices.py [hunk 3]
```diff
@@ -2000,7 +1999,8 @@ class PositionalOnlyArgumentsViolation(ASTViolation):
 
     Solution:
         Use regular arguments.
-        In case you are working with C, then this violation can be ignored.
+        In case you are working with C programming language,
```
[the remark this answers is not in the document]
Comment (swapniljha001) [reply wemake_python_styleguide/violations/best_practices.py:2002]: Done.

File: wemake_python_styleguide/violations/best_practices.py [hunk 4]
```diff
@@ -2075,15 +2075,15 @@ class ShebangViolation(SimpleViolation):
         - Shebang is present but the file is not executable.
         - The file is executable but no shebang is present.
         - Shebang is present but does not contain "python".
-        - There is whitespace before shebang.
-        - There are blank or comment lines before shebang.
+        - Presence of whitespace before the shebang.
+        - Presence of blank lines or commented lines before the shebang.
```
[the remark this answers is not in the document]
Comment (swapniljha001) [reply wemake_python_styleguide/violations/best_practices.py:2079]: Done

File: wemake_python_styleguide/violations/best_practices.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (swapniljha001) [reply wemake_python_styleguide/violations/best_practices.py:2085]: Fixed.

File: wemake_python_styleguide/violations/best_practices.py [hunk 5]
```diff
@@ -2075,15 +2075,15 @@ class ShebangViolation(SimpleViolation):
...
 
     Reasoning:
         Setting the shebang incorrectly causes an executable mismatch.
 
     Solution:
-        Ensure the shebang is present on the first line,
-        contains "python", and there is no whitespace before.
+        Ensure that the shebang present on the first line,
+        contains "python", and there is no whitespace beforehand.
```
[the remark this answers is not in the document]
Comment (swapniljha001) [reply wemake_python_styleguide/violations/best_practices.py:2086]: Done.

File: wemake_python_styleguide/violations/best_practices.py [hunk 6]
```diff
@@ -1958,10 +1958,9 @@ class ProtectedModuleMemberViolation(ASTViolation):
     Related to :class:`~ProtectedModuleViolation`.
 
     Reasoning:
-        When importing protected modules' members we break a contract
-        that authors of this module enforce.
-        This way we are not respecting encapsulation and it may break
-        our code at any moment.
+        When importing protected modules' members, we break the contract
+        which the authors of this module enforce.
+        Thus, disrespecting encapsulation may break the code at any moment.
```
[the remark this answers is not in the document]
Comment (swapniljha001) [reply wemake_python_styleguide/violations/best_practices.py:1963]: Done.

File: wemake_python_styleguide/violations/best_practices.py [hunk 7]
```diff
@@ -1884,7 +1884,7 @@ class IncorrectExceptOrderViolation(ASTViolation):
 
     Reasoning:
         Using incorrect order of exceptions is error-prone, since
-        you end up with some unreachable exception clauses.
+        once ends up with some unreachable exception clauses.
```
[the remark this answers is not in the document]
Comment (swapniljha001) [reply wemake_python_styleguide/violations/best_practices.py:1887]: Changed it to "you" as earlier for consistency.

**Comment by swapniljha001** [comment]
I have made the suggested changes and will try to look for other grammatical mistakes.

**Comment by Dreamsorcerer** [comment]
@sobolevn Looks good to me. Let's merge this, and put any other changes in a new PR?

**Review by sobolevn — APPROVED** [review]

**Comment by sobolevn** [comment]
Thanks a lot, @Dreamsorcerer and @swapniljha001

**Pull request merged**